### PR TITLE
feat: filter fields supported by all objects

### LINKED ISSUE #7320: feat: new API design for the Admin API
### Description

One of the plans in the APISIX V3 roadmap was discussed in the mailing list: new API design for the Admin API

mail list link: https://lists.apache.org/thread/jg46m1lxn2zx3gz1r77b3v6cz6v2l4bn

Now I'm going to implement it, and this task is split into the following subtasks:

- [ ] updated some list URIs(SSL, proto )
- [ ] response body format
- [ ] supports paging to get a list of objects such as route
- [ ] supports filter when getting a list of objects such as route
- [ ] additional filter fields supported by the Route object

This issue is used to track the progress of this task, and also used to record things that require collaboration with other projects (dashboard, apisix-ingress-controller)

### PR DESCRIPTION
**Pull request by tzssangglass**
### Description

Fixes #7320

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Review by membphis — APPROVED** [review]

File: apisix/admin/v3_adapter.lua [hunk 1]
```diff
@@ -122,4 +118,55 @@ function _M.filter(body)
...
+        local uri_matched = true
+        if args.name then
+            local matched = re_find(body.list[i].value.name, args.name, "jo")
+            if not matched then
+                name_matched = false
+            end
+        end
+
+        if args.label then
+            local matched = re_find(body.list[i].value.label, args.label, "jo")
```
Comment (spacewander) [inline apisix/admin/v3_adapter.lua:138]: We should match one of the labels?
https://github.com/apache/apisix/blob/a8a692d9a58a9a0185e22cfebbe2c0f000ba41f3/apisix/schema_def.lua#L693

Comment (tzssangglass) [reply apisix/admin/v3_adapter.lua]: update

File: apisix/admin/v3_adapter.lua [hunk 2]
```diff
@@ -135,9 +136,24 @@ local function filter(body, args)
         end
 
         if args.label then
-            local matched = re_find(body.list[i].value.label, args.label, "jo")
-            if not matched then
-                label_matched = false
+            if body.list[i].value.label then
```
Comment (spacewander) [inline apisix/admin/v3_adapter.lua:139]: Is there somewhere we use label to mark a resource? AFAIK, we only have `labels`, no `label`

**Review by spacewander — APPROVED** [review]

**Review by soulbird — APPROVED** [review]

File: apisix/admin/v3_adapter.lua (hunk 2 above)
Comment (tzssangglass) [reply apisix/admin/v3_adapter.lua]: updated

**Review by spacewander — APPROVED** [review]

**Pull request merged**